src/plugins/intel_gpu/include/intel_gpu/runtime/lru_cache.hpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include <list>
+#include <unordered_map>
+#include <functional>
+#include <iostream>
+
+namespace cldnn {
+
+struct primitive_impl;
+
+/// @brief LRU cache which remove the least recently used data when cache is full.
+template<typename TypeK, typename TypeD>
+class LRUCache {
+public:
+    struct CacheEntry {
+        TypeD   data;
+        size_t  size;   // the size of data. it is necessary to check total data size is limited to capacity
```

Comment:
When I design `impl_cache`, I made cache entry to check the limit of capacity with both number of entry and byte size. But I think we don't need to check byte size to check the limit of cap now. So I will modify code to use `TypeD `instead of Cache Entry by your comments. thanks.

Reply:
I removed curr_data_size and CacheEntry. Could you please review it again?